cborg/tests/Tests/ByteOffset.hs

```diff
@@ -63,6 +66,24 @@ prop_ATerm_isomorphic3 t =
 -- Properties of peekByteOffset
 --
 
+-- | A smoke test that nothing weird happens with an empty span the input.
+empty_deserialise :: Property
+empty_deserialise = res === (0, 0)
+  where
+    -- unfortunately we need a non-empty input stream.
+    -- the decoder will fail unconditionally, if there are commands,
+    -- even ones which consume nothing.
+    res = either throw snd (deserialiseFromBytes emptyDecoder (LBS.pack [0]))
+
+empty_deserialise_fail :: Bool
+empty_deserialise_fail = isLeft (deserialiseFromBytes emptyDecoder LBS.empty)
```

Comment:
I added these tests for both `peekByteOffset` and `getInputSpan`. 

I find it a bit surprising as `peekByteOffset` as well as `unmarkInput` and `getInputSpan` do succeed at the end of input, but they seem to fail at the very beginning of an empty input.

This is quite obscure corner case, but nevertheless I think it's good to be aware off (i.e. have test for it).